src/v0/destinations/active_campaign/util.js

```diff
@@ -1,19 +1,24 @@
 const {
   nodeSysErrorToStatus
 } = require("../../../adapters/utils/networkUtils");
-const { CustomError } = require("../../util");
+const {
+  RetryableError,
+  NetworkInstrumentationError
+} = require("../../util/errorTypes");
 
 const errorHandler = (err, message) => {
   if (err.response) {
-    throw new CustomError(
+    throw new NetworkInstrumentationError(
       `${message} (${err.response?.statusText},${JSON.stringify(
         err.response?.data
-      )})`,
-      err.response.status || 400
+      )})`
     );
   } else {
     const httpError = nodeSysErrorToStatus(err.code);
-    throw new CustomError(`${message} ${httpError.message}`, httpError.status);
+    if (httpError.status === 500) {
+      throw new RetryableError(`${message} ${httpError.message}`);
+    }
+    throw new NetworkInstrumentationError(`${message} ${httpError.message}`);
```

Comment:
Is this accurate w.r.t the docs?

Reply:
In `networkUtils`, we have a mapping of `ReceivedCode-> status`, for 500 status we can do retryable. We are returning 400 [here](https://github.com/rudderlabs/rudder-transformer/blob/23ce982f2756c62d009250aba3cafc87f4b63b67/src/adapters/utils/networkUtils.js#L15) in multiple scenarios.